### ISSUE TITLE
3_Open_Source_Software.md - #cost

### ISSUE THREAD

**Issue by gcharest** [canada-ca/Open_First_Whitepaper#81]
In [3_Open_Source_Software.md#cost](https://github.com/canada-ca/Open_First_Whitepaper/blob/master/3_Open_Source_Software.md#cost), I believe the following statement does not necessarily fully convey the reality that implementation of FLOSS solutions may also have considerable costs outside the licensing aspects:

> When using OSS, the up-front licensing cost is zero (aside from internal costs of assessing the software and the licence). A OSS licence is always “open” and granted to the world, in that it licenses everyone to use the software without any fees due. Moreover, due to the fact that OSS licences grant everyone the right to redistribute the software, nearly all OSS packages are available for free download over the internet.
> As well, OSS software often incurs lower on-going maintenance costs. With closed-source software, the vendor and its select business partners are quite often the only businesses able to provide adequate support (either because the software licence grants the vendor an exclusive support contract, or due to the vendor's specialized expertise with the software and its sole ability to examine and work with the source code).

The problem I see is that implementation cost seems to be glanced over in the paragraph above. For example, packaging and providing Firefox as a browser option for corporate users still has a cost of implementation although it is extremely lower than setting up a corporate website using Drupal. One is mature and "complete" in itself, the other needs a series of integration with other components as well as configuration and customization.

And further down:

> The total cost of ownership is shared with a community, rather than solely the Government of Canada.

Again, yes and no. The cost of development and maintenance of the core software may be shared but the implementation may still require professional services at the expense of the GoC (Drupal comes to mind again and we can chat with the Open Gov Pilot team for that).

Just thought I'd bring this to the group for discussion.

**Comment by mgifford** [comment]
1) I guess it comes down to different software has different implementation costs and this is the same for all software. Something like FF requires no customization to be useful, where almost nobody runs their website with Drupal Core alone (it's certainly possible with Drupal 8, but nobody does it). This will be the same for most desktop applications vs websites though. 

2) It wasn't quite where you were going, but I wanted to add critical mass to this point. Only when there is a critical mass of users is the cost truly supported by a broader community. That's one reason that Drupal stands out, is that it has that critical mass of professional web shops around the world that are continuing to push the product ahead. Custom code developed by government departments and released as open source likely won't grow to have a critical mass of users outside that department. It might, but you can't depend on it. 

And yes, even when it comes to installing Firefox on desktops across the GoC, it will require support time within government to implement and maintain it. With more complex custom development like Drupal, it will take a more sophisticated team (inside of government and possibly outside too). But this is the case for any successful customized software application. Hopefully the days of government hiring an IT firm and just assuming that they magically implement what they need are over.  Government needs people on the inside that understand open source, accessibility and security.

**Comment by gcharest** [comment]
@mgifford I agree with you regarding the critical mass for actual support from the community. I think my point here is to make sure we don't take the mental shortcut of equating free license with free solution. The team working on the Open Gov Pilot project have opened my eyes on this!

On the other end, it does mean breaking lock-in situations and enabling more companies to offer their professional services to the GoC.

**Comment by mgifford** [comment]
Totally, free as in kittens.

**Comment by Acasovan** [comment]
Thanks @gcharest as the former business owner of open.canada.ca, I'm fully aware of the cost issues! That's why I became so interested in working on this project. Tackling how we assess the cost and fund appropriately is one of our key objectives with implementing a policy standard. 

That said, we know this section needs lots of work. I'll look at updating this sections as soon as I get a chance. A lot more work in general needs to be done on the cost of ownership, which I think involves research that we haven't done to this point. We can only use the projects that we have used open source for to this point, but project costs can vary drastically. 

Have added this to the critical to-do-list.

**Comment by gcharest** [comment]
Hi @Acasovan !

I'll try to reach out to a former University professor of mine. He had done research on TCO for OSS solutions.

Will keep you posted!

**Comment by JasonBee** [comment]
Looking for another participant in government? 

I have some detailed comments on the OS vs CS discussion though more on the integration costs and value calculations, if still relevant. It all depends  great deal on how well the orgs have implemented operational frameworks, efficiency and effectiveness at training and development, level of friction for internal collaboration and knowledge sharing, competing priorities etc.

It would be nice to followup on that TCO point...was there any further discussion?

**Comment by gcharest** [comment]
@JasonBee I've received very interesting evaluation grids from the Ville de Montreal but I have not yet had the chance to update this section.

I'm open for input as we need to wrap up version 1.0.
